File: typescript/src/models/oAuthErrorResponse.ts
// this file is @generated
import { type OAuthErrorCode, OAuthErrorCodeSerializer } from "./oAuthErrorCode";
/** OAuth 2.0 error response as per RFC 6749 Section 5.2 */
export interface OAuthErrorResponse {
  error: OAuthErrorCode;

  errorDescription?: string | null;

  errorUri?: string | null;
}

export const OAuthErrorResponseSerializer = {
  _fromJsonObject(object: any): OAuthErrorResponse {
    return {
      error: OAuthErrorCodeSerializer._fromJsonObject(object["error"]),
      errorDescription: object["error_description"],
      errorUri: object["error_uri"],
    };
  },

  _toJsonObject(self: OAuthErrorResponse): any {
    return {
      error: OAuthErrorCodeSerializer._toJsonObject(self.error),
      error_description: self.errorDescription,
      error_uri: self.errorUri,
    };
  },
};

File: rust/src/models/o_auth_error_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::o_auth_error_code::OAuthErrorCode;

/// OAuth 2.0 error response as per RFC 6749 Section 5.2
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct OAuthErrorResponse {
    pub error: OAuthErrorCode,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub error_description: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub error_uri: Option<String>,
}

impl OAuthErrorResponse {
    pub fn new(error: OAuthErrorCode) -> Self {
        Self {
            error,
            error_description: None,
            error_uri: None,
        }
    }
}
